src/main/webapp/components/TopNavigationMenu.css

```diff
@@ -0,0 +1,94 @@
+/* Logo */
+.navbar-default .navbar-brand:active,
+.navbar-default .navbar-brand:focus,
+.navbar-default .navbar-brand:hover {
+    color: #FFFFFF;
+    -webkit-transition: background-color 0.3s ease-in-out;
+  -moz-transition: background-color 0.3s ease-in-out;
+  -o-transition: background-color 0.3s ease-in-out;
+  transition: background-color 0.3s ease-in-out;
+}
+
+.navbar-default .navbar-brand:hover {
+    background-color: #39506A;
+}
+
+/* Navigation */
+.navbar-default .navbar-nav>li>a:active,
+.navbar-default .navbar-nav>li>a:focus {
+    color: #FFFFFF;
+}
+
+.navbar-form .select__indicators {
+    display: none;
+    visibility: hidden;
+}
+
+.navbar .googleButton {
+    float: right;
+    overflow: hidden;
+    margin-top: 5px;
+}
+
+.navbar .googleButton button,
+.navbar .googleButton button div,
+.navbar .googleButton button span {
+    height: 40px;
+}
+
+/* Search */
+.navbar-default .navbar-form.navbar-left {
+    width: 220px;
+    display: inline-block;
+    margin: 3px 0 4px 0;
+}
+
+/* Search Results (list) */
+
+.navbar .inputField .inputFieldSelect .select__menu .select__menu-list {
+    width: 500px;
+    z-index: 3000;
+    max-height: 400px;
+    background-color: #FFFFFF;
+    overflow-y: auto;
+    position: fixed;
+    top: 52px;
+    border: 1px solid #CCCCCC;
+    border-radius: 4px;
+}
+
+.navbar .inputField .inputFieldSelect .select__control {
+    border-radius: 4px;
+    padding: 5px 8px;
+    height: 40px;
+}
+
+.navbar .inputField .inputFieldSelect .select__control .select__value-container {
+    margin-top: -3px;
+}
+
+.navbar .inputFieldSelect {
+    padding: 0;
+    margin: 0;
+}
+
+.navbar .inputField .inputFieldSelect .select__option {
+    background-color: #FFFFFF;
+    cursor: pointer;
+    padding: 0;
+    display: block;
+    margin: 0;
+}
+
+.navbar .inputField .inputFieldSelect .select__option.select__option--is-focused:hover {
+    color: #337ab7;
+}
+
+.navbar .inputField .inputFieldSelect .select__option.select__option--is-disabled {
+    background-color: #FAFAFA;
+    padding: 15px !important;
+}
+
+.navbar .inputField .inputFieldSelect .select__menu .select__menu-list .select__option {
+    padding: 20px 15px;
+}
```

Comment:
The problem I'm having with inline styling are class names (such as: ".select__menu-list") and how they are created/rendered, they are auto-generated components and we aren't assigning them className or style directly. Is for that reason I'm editing appearance via CSS, because I'm not getting access to them via inline. Also the chain of concatenated classes is a problem to me for reproducing it inline. Any help is welcome

Reply:
We *should* be assigning the styles directly, that's the point of having inline styles. First, you create the style object, then you use that object as a component property, i.e. `ComponentX({style: {property: "value"}});` or `ComponentX({style: styleObject.myComponent});`

Shared CSS requires far more bug checking than inline styles. If I make a change to a classname css, then I need to physically view all usages of that css to ensure no bugs were introduced. Isolating the CSS to the component that uses it reduces the number of bugs introduced.

Reply:
@rushtong Search list styles are now included in the TopNavigationMenu.js and removed from CSS file